sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -581,39 +552,44 @@ Member Functions are described in the table below
 a|
 [source,c++]
 ----
-annotated_ref(const annotated_ref&) = default;
+operator T() const;
 ----
 |
-Compiler generated copy constructor.
+Reads the object of type `T` that is referenced by this wrapper, applying the
+annotations when the object is loaded from memory.
 
 // --- ROW BREAK ---
 a|
 [source,c++]
 ----
-operator T() const;
+T operator=(const T &);
 ----
 |
-Reads the object of type `T` that is referenced by this wrapper, applying the
-annotations when the object is loaded from memory.
+Writes an object of type `T` to the location referenced by this wrapper,
+applying the annotations when the object is stored to memory.
 
 // --- ROW BREAK ---
 a|
 [source,c++]
 ----
-annotated_ref& operator=(const T &);
+T operator=(const annotated_ref&);
```

Comment:
I'm not sure we want this.  Before this PR, this assignment operator just assigned one `annotated_ref` to another without accessing the underlying memory.  What was the rationale for removing this (and the associated copy constructor)?

The new semantic added by this PR seems wrong to me.  I don't think we want assignment to another `annotated_ref` to cause a store to "this" memory.  If it has that semantic, doesn't that mean there is also a load from "other" memory?

Looking at C++ `atomic_ref`, I see that they do not have `operator=(const atomic_ref &)`, so maybe we should just delete this operator in `anotated_ptr`.  Oddly, I see that `atomic_ref` *does* have a copy constructor `atomic_ref(const atomic_ref &)`.  Should we add that copy constructor back to `annotated_ref`?  Why does `atomic_ref` have this inconsistency between the copy constructor and the copy-assignment operator?

Reply:
Yes, that's a good point.  Supporting `*p1 = *p2` does seem natural.  Let's make the description clearer, though, to specify how the annotations work.  For example:

> `T operator=(const annotated_ref& other)`
>
> Equivalent to:
> ```
> tmp T = other;   // Reads from memory while applying annotations from "other"
> *this = T;       // Writes to memory while applying annotations from "this"
> return T;
> ```
> Does not rebind the reference!